File: src/java/magmac/app/lang/node/Constructor.java
package magmac.app.lang.node;

import magmac.api.Option;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;
import magmac.app.lang.Destructors;

public final class Constructor implements JavaMethodHeader, TypeScriptMethodHeader {
    public static Option<CompileResult<JavaMethodHeader>> deserialize(Node node) {
        return Destructors.destructWithType("constructor", node)
                .map(constructor -> constructor.complete(Constructor::new));
    }

    @Override
    public Node serialize() {
        return new MapNode("constructor");
    }
}

File: src/web/magmac/app/lang/node/Constructor.ts
import { Option } from "../../../../magmac/api/Option";
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { MapNode } from "../../../../magmac/app/compile/node/MapNode";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { Deserializers } from "../../../../magmac/app/lang/Deserializers";
export class Constructor {
	deserialize(node : Node) : Option<CompileResult<JavaMethodHeader>>;
	serialize() : Node;
}
